src/sail_json_backend/json.ml

```diff
@@ -231,20 +232,63 @@ let parse_encdec i mc format =
       end
   | _ -> assert false
 
+let rec filter_non_operands components =
+  match components with
+  | [] -> []
+  | hd :: tl when String.trim hd = "spc" -> tl
+  | _ -> filter_non_operands (List.tl components)
+
+let rec extract_operands k filtered_components =
+  match filtered_components with
+  | [] -> []
+  | hd :: tl ->
+      if Str.string_match (Str.regexp ".+(\\(.*\\))") hd 0 then (
```

Comment:
For somebody new to the code, this is a diversion from just looking for operands -- due to constructs in the `mapping clause assembly` where an operand is within a function call, like `nzimm` in:
```
"c.addi4spn" ^ spc() ^ creg_name(rdc) ^ sep() ^ hex_bits_10(nzimm @ 0b00)
```

So, let's add a comment here, perhaps something like:
> Look for operands embedded within functions like "funct(op1 @ op2 @ 0b00)"